### ISSUE TITLE
CC1101 Pager RX

### ISSUE THREAD

**Issue by joelsernamoreno** [jgromes/RadioLib#1261]
Hi, I am using ESP32 and CC1101 for TX and RX. The TX code is as follows:

<details><summary>Details</summary>
<p>

```c++
#include <RadioLib.h>

CC1101 radio = new Module(5, 2, RADIOLIB_NC, 4);
PagerClient pager(&radio);

void setup() {
  Serial.begin(115200);
  delay(7000);
  Serial.print("[ESP32] Initializing ... ");
  int state = radio.begin();

  if(state == RADIOLIB_ERR_NONE) {
    Serial.println("success!");
  } else {
    Serial.print("failed, code ");
    Serial.println(state);
    while (true) { delay(10); }
  }

  Serial.print("[Pager] Initializing ... ");
  state = pager.begin(434.0, 1200);

  if(state == RADIOLIB_ERR_NONE) {
    Serial.println("success!");
  } else {
    Serial.print("failed, code ");
    Serial.println(state);
    while (true) { delay(10); }
  }
}

void loop() {
  Serial.print("[Pager] Transmitting messages ... ");
  int state = pager.sendTone(1234567);
  delay(500);
  state |= pager.transmit("0123456789*U -()", 1234567);
  delay(500);
  state |= pager.transmit("Test", 1234567, RADIOLIB_PAGER_ASCII);
  delay(500);

  if(state == RADIOLIB_ERR_NONE) {
    Serial.println("success!");
  } else {
    Serial.print("failed, code ");
    Serial.println(state);
  }
  delay(10000);
}
```

</p>
</details> 

I can see with an SDR the transmitted signal, but there is something strange here. I think the signal is not 2-FSK.

Additionally, this is my RX code, I do not receive the TX message:

<details><summary>Details</summary>
<p>

```c++
#include <RadioLib.h>

CC1101 radio = new Module(5, 2, RADIOLIB_NC, 4);
const int pin = 4;
PagerClient pager(&radio);

void setup() {
  Serial.begin(115200);
  delay(5000);
  Serial.print("[ESP32] Initializing ... ");
  int state = radio.begin();

  if (state == RADIOLIB_ERR_NONE) {
    Serial.println("success!");
  } else {
    Serial.print("failed, code ");
    Serial.println(state);
    while (true) { delay(10); }
  }

  Serial.print("[Pager] Initializing ... ");
  state = pager.begin(434.0, 1200);

  if (state == RADIOLIB_ERR_NONE) {
    Serial.println("success!");
  } else {
    Serial.print("failed, code ");
    Serial.println(state);
    while (true) { delay(10); }
  }

  Serial.print("[Pager] Starting to listen ... ");
  state = pager.startReceive(pin, 1234567);

  if (state == RADIOLIB_ERR_NONE) {
    Serial.println("success!");
  } else {
    Serial.print("failed, code ");
    Serial.println(state);
    while (true) { delay(10); }
  }
}

void loop() {
  if (pager.available() >= 2) {
    Serial.print("[Pager] Received pager data, decoding ... ");

    String str;
    int state = pager.readData(str);

    if (state == RADIOLIB_ERR_NONE) {
      Serial.println("success!");

      Serial.print("[Pager] Data:\t");
      Serial.println(str);

    } else {
      Serial.print("failed, code ");
      Serial.println(state);

    }
  }
}
```

</p>
</details> 

Thanks

**Comment by Szetya** [comment]
> Hi, I am using ESP32 and CC1101 for TX and RX. The TX code is as follows:
> 
> Details
> I can see with an SDR the transmitted signal, but there is something strange here. I think the signal is not 2-FSK.
> 
> Additionally, this is my RX code, I do not receive the TX message:
> 
> Details
> Thanks

Please try version 5.7.0 for both TX and RX.

**Comment by joelsernamoreno** [comment]
> > Hi, I am using ESP32 and CC1101 for TX and RX. The TX code is as follows:
> > Details
> > I can see with an SDR the transmitted signal, but there is something strange here. I think the signal is not 2-FSK.
> > Additionally, this is my RX code, I do not receive the TX message:
> > Details
> > Thanks
> 
> Please try version 5.7.0 for both TX and RX.

Hello @Szetya 

I have used RadioLib Patch 7.0.2 for TX and RadioLib 5.6.0 for RX. This is working.

Do you need me to try 5.7.0?

**Comment by Szetya** [comment]
> > > Hi, I am using ESP32 and CC1101 for TX and RX. The TX code is as follows:
> > > Details
> > > I can see with an SDR the transmitted signal, but there is something strange here. I think the signal is not 2-FSK.
> > > Additionally, this is my RX code, I do not receive the TX message:
> > > Details
> > > Thanks
> > 
> > 
> > Please try version 5.7.0 for both TX and RX.
> 
> Hello @Szetya
> 
> I have used RadioLib Patch 7.0.2 for TX and RadioLib 5.6.0 for RX. This is working.
> 
> Do you need me to try 5.7.0?

Hi @joelsernamoreno 
In my experience, the last version that works on TX and RX is 5.7.0 
That's why I asked you to try it if you have the chance.

**Comment by joelsernamoreno** [comment]
> > > > Hi, I am using ESP32 and CC1101 for TX and RX. The TX code is as follows:
> > > > Details
> > > > I can see with an SDR the transmitted signal, but there is something strange here. I think the signal is not 2-FSK.
> > > > Additionally, this is my RX code, I do not receive the TX message:
> > > > Details
> > > > Thanks
> > > 
> > > 
> > > Please try version 5.7.0 for both TX and RX.
> > 
> > 
> > Hello @Szetya
> > I have used RadioLib Patch 7.0.2 for TX and RadioLib 5.6.0 for RX. This is working.
> > Do you need me to try 5.7.0?
> 
> Hi @joelsernamoreno In my experience, the last version that works on TX and RX is 5.7.0 That's why I asked you to try it if you have the chance.

Hi @Szetya 

I have tested 5.7.0 for TX and RX, it does not work for me. I think the TX power with this version is too small, even if I bring the antennas closer I don't get the message. With an SDR I can see the signal, but this is very low power

With Patch 7.0.2 in TX it works correctly

**Comment by Szetya** [comment]
> > > > > Hi, I am using ESP32 and CC1101 for TX and RX. The TX code is as follows:
> > > > > Details
> > > > > I can see with an SDR the transmitted signal, but there is something strange here. I think the signal is not 2-FSK.
> > > > > Additionally, this is my RX code, I do not receive the TX message:
> > > > > Details
> > > > > Thanks
> > > > 
> > > > 
> > > > Please try version 5.7.0 for both TX and RX.
> > > 
> > > 
> > > Hello @Szetya
> > > I have used RadioLib Patch 7.0.2 for TX and RadioLib 5.6.0 for RX. This is working.
> > > Do you need me to try 5.7.0?
> > 
> > 
> > Hi @joelsernamoreno In my experience, the last version that works on TX and RX is 5.7.0 That's why I asked you to try it if you have the chance.
> 
> Hi @Szetya
> 
> I have tested 5.7.0 for TX and RX, it does not work for me. I think the TX power with this version is too small, even if I bring the antennas closer I don't get the message. With an SDR I can see the signal, but this is very low power
> 
> With Patch 7.0.2 in TX it works correctly

Thank you for the test. Maybe this is also an information for Jgromes.

**Comment by joelsernamoreno** [comment]
Hello @Szetya @jgromes 

I have tested the RX example with TX portapack mayhem (hackrf). The POCSAG module of hackrf transmits a message and I can receive this on ESP32+CC1101.

If I try to transmit a POCSAG message with ESP32+CC1101 I can receive it on another ESP32+CC1101, on the other hand, if I transmit the message with ESP32+CC1101... portapack mayhem (hackrf) does not receive this. Alphapoc pager does not receive the message either

Any hint? I have set the frequency, ID and baud rate correctly.
